analysis/bin/doubleHiggsAnalyser.cc

```diff
@@ -0,0 +1,212 @@
+void doubleHiggsAnalyser::SetOutput(TString output_file_name) {
+  out_file = TFile::Open(output_file_name,"RECREATE");
+  out_tree = new TTree("events","events"); 
+}
+
+void doubleHiggsAnalyser::SetBranchAddress() {
+  del_tree->SetBranchAddress("Particle",&particles);
+  del_tree->SetBranchAddress("MissingET",&missings);
+  del_tree->SetBranchAddress("Jet",&jets);
+}
+
+void doubleHiggsAnalyser::Initiate(TString output_file_name) {
+  // set output file
+  doubleHiggsAnalyser::SetOutput(output_file_name);
+  // make output branch
+  doubleHiggsAnalyser::MakeOutputBranch(out_tree);
+  // set branch address
+  doubleHiggsAnalyser::SetBranchAddress();
+}
+
+void doubleHiggsAnalyser::ResetVariables() {
+  lester_MT2 = -99;
+  basic_MT2_332 = -99;
+  ch_bisect_MT2_332 = -99;
+  lepton1_pt = -99;
+  lepton2_pt = -99;
+  missing_et = -99;
+  mt = -99;
+  fromHiggs = 0;
+  fromTop = 0;
+  fromZ = 0;
+  from1 = 0;
+  from2 = 0;
+  
+  muons.clear();
+  bottoms.clear();
+}
+
+bool doubleHiggsAnalyser::Analysis() {
+  //map<Float_t, int, greater<Float_t>> muons : map of <pt,index>:<K,V> of muon sorted by pt.
+  //base selections : MissingET > 20, pT(lepton) > 20, deltaR(ll) < 1.0, m(ll) < 65, deltaR(bb) < 1.3, 95 < m(bb) < 140
+    auto m = static_cast<const MissingET *>(missings->At(0)); // There is always one MET object.
+    if (m->MET<20) return false;
+    missing.SetPtEtaPhiM(m->MET,0,m->Phi,0);
+ 
+    bool jetflag = false;
+    for (int ij = 0; ij < jets->GetEntries(); ij++){
+      auto jet = static_cast<const Jet *>(jets->At(ij));
+      if (abs(jet->Flavor) != 5) continue;
+      else {jetflag = true; break;}
+    }
+    if (!jetflag) return false;
+    
+    
+    for (int ip = 0; ip < particles->GetEntries(); ip++){
+      auto p = static_cast<const GenParticle *>(particles->At(ip));
+      if (abs(p->PID)!=13 || p->PT<20 || fabs(p->Eta)>2.5 || p->M1==-1) continue;
+      muons.insert(make_pair(p->PT,ip));
+    }
+  
+    if (muons.size()<2) {
+      return false;
+    }
+    
+    map<Float_t,int,greater<Float_t>>::iterator iter = muons.begin();
+    auto mu1 = static_cast<const GenParticle *>(particles->At(iter->second));
+    int from1 = isFrom(particles, iter->second);
+    from1 = abs(from1);
+    if (from1==25) fromHiggs++;
+    if (from1==6) fromTop++;
+    if (from1==32) fromZ++;
+    muon1.SetPtEtaPhiM(mu1->PT,mu1->Eta,mu1->Phi,mu1->Mass);
+    ++iter;
+    auto mu2 = static_cast<const GenParticle *>(particles->At(iter->second));
+    int from2 = isFrom(particles, iter->second);
+    from2 = abs(from2);
+    if (from2==25) fromHiggs++;
+    if (from2==6) fromTop++;
+    if (from2==32) fromZ++;
+    muon2.SetPtEtaPhiM(mu2->PT,mu2->Eta,mu2->Phi,mu2->Mass);
+    
+    lepton1_pt = mu1->PT;
+    lepton2_pt = mu2->PT;
+    missing_et = m->MET;
+    auto muons = muon1+muon2;
+    if (muons.M() > 65 || muon1.DeltaR(muon2) > 1) {
+        return false;
+    }
+    mt = muons.M()*muons.M() + muons.Px()*muons.Px() + muons.Py()*muons.Py();
+
```

Comment:
isnt this the same as mouns.Mt()?

Reply:
I thought so ... but they weren't same so I was trying to see the distribution for each of them.

Reply:
you should be doing the sqrt of this